packages/jupyter-ai/jupyter_ai/auth/identity.py

```diff
@@ -0,0 +1,39 @@
+import asyncio
+import getpass
+
+from jupyter_server.auth.identity import IdentityProvider, User
+
+
+def create_initials(username):
+    """Creates initials combining first 2 consonants"""
+
+    username = username.lower()
+
+    # Default: return first two unique consonants
+    consonants = [c for c in username if c in "bcdfghjklmnpqrstvwxyz"]
+    if len(consonants) >= 2:
+        return (consonants[0] + consonants[1]).upper()
+
+    # Fallback: first two characters
+    return username[:2].upper()
+
+
+class LocalIdentityProvider(IdentityProvider):
+    """IdentityProvider that determines username from system user."""
+
+    def get_user(self, handler):
+        try:
+            username = getpass.getuser()
+            user = User(
+                username=username,
+                name=username,
+                initials=create_initials(username),
+                color=None,
+            )
+            return user
+        except OSError as e:
```

Comment:
nit: not using variable `e`, interesting that the linter isn't picking this up...